datasync/chaindatafetcher/kafka/kafka_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kafka_test
+
+import (
+	"encoding/json"
+	"math/rand"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/klaytn/klaytn/datasync/chaindatafetcher/kafka"
+	"github.com/klaytn/klaytn/datasync/chaindatafetcher/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type kafkaData struct {
+	Name   string `json:"name"`
+	Data   string `json:"data"`
+	Number int    `json:"number"`
+	IsGood bool   `json:"isGood"`
+}
+
+type KafkaSuite struct {
+	suite.Suite
+	kfk   types.EventBroker
+	topic string
+	data  string
+}
+
+func (s *KafkaSuite) SetupSuite() {
+	s.topic = "test-topic"
+	s.data = "test-data"
+	s.kfk = kafka.New("test_group_id", []string{"kafka:9094"}, 1, 1)
+}
+
+func (s *KafkaSuite) TearDownSuite() {
+	// Assume that the delete topic works properly.
+	s.kfk.DeleteTopic(s.topic)
+}
+
+func (s *KafkaSuite) TestKafka_CreateAndDeleteTopics() {
+	// no topic to be deleted
+	err := s.kfk.DeleteTopic(s.topic)
+	s.Error(err)
+	s.True(strings.Contains(err.Error(), sarama.ErrUnknownTopicOrPartition.Error()))
+
+	// created a topic successfully
+	topic, err := s.kfk.CreateTopic(s.topic)
+	s.NoError(err)
+	s.Equal(s.topic, topic.Name)
+
+	// failed to create a duplicated topic
+	duplicatedTopic, err := s.kfk.CreateTopic(s.topic)
+	s.Error(err)
+	s.True(strings.Contains(err.Error(), sarama.ErrTopicAlreadyExists.Error()))
+	s.Equal(topic, duplicatedTopic)
+
+	// deleted a topic successfully
+	s.Nil(s.kfk.DeleteTopic(s.topic))
+}
+
+func (s *KafkaSuite) TestKafka_Success_PubSub() {
+	numTests := 10
+	done := make(chan *kafkaData, numTests)
+	s.kfk.CreateTopic(s.topic)
+
+	// subscribe the test topic to check the data correctness
+	err := s.kfk.Subscribe(s.topic, func(msg *sarama.ConsumerMessage) error {
+		s.Equal(s.topic, msg.Topic)
+		d := &kafkaData{}
+		s.Nil(json.Unmarshal(msg.Value, d))
+		done <- d
+
+		return nil
+	})
+	s.NoError(err)
+
+	// sleep for a while for the subscription to be set
+	time.Sleep(3 * time.Second)
```

Comment:
@jeongkyun-oh Could you reduce the sleep time here and other places? It will increase CircleCi testing time.

Reply:
I reduced them. But, the more time is taken for the other packages for `test-datasync` for your information.

![스크린샷 2020-09-23 오전 11 12 45](https://user-images.githubusercontent.com/45347815/93957230-d04b4a80-fd8e-11ea-97b0-937970fccd4a.png)

Reply:
thanks!